iterator.go

```diff
@@ -0,0 +1,63 @@
+package cdb
+
+// Iterator represents an iterator over a CDB database. It is used to sequentially
+// iterate over all kv-pairs in the database.
+type Iterator struct {
+	db     *CDB
+	pos    uint32
+	endPos uint32
+	err    error
+	key    []byte
+	value  []byte
+}
+
+// Iter creates an Iterator that can be used to iterate the database.
+func (cdb *CDB) Iter() *Iterator {
+	return &Iterator{
+		db:     cdb,
+		pos:    uint32(indexSize),
+		endPos: cdb.index[0].offset,
+	}
+}
+
+// Next reads the next key-value pair and advances the iterator one record.
+// If Next returns true it successfully updated the current key/value
+// and has more records to iterate over.
+// If Next returns false it either encountered an error (can be read through Err())
+// or it has no more records to iterate over.
+func (iter *Iterator) Next() bool {
+	keyLength, valueLength, err := readTuple(iter.db.reader, iter.pos)
+	if err != nil {
+		iter.err = err
+		return false
+	}
+
+	buf := make([]byte, keyLength+valueLength)
+	_, err = iter.db.reader.ReadAt(buf, int64(iter.pos+8))
+	if err != nil {
+		iter.err = err
+		return false
+	}
+
+	// Update iterator state
+	iter.key = buf[:keyLength]
+	iter.value = buf[keyLength:]
+	iter.pos += 8 + keyLength + valueLength
+
+	return iter.pos < iter.endPos
+}
+
+// Key returns the current key.
```

Comment:
We should add something here (and to `Value`) like "The byte slice returned is only valid until the next call to Next."